dotCMS/src/main/java/com/dotmarketing/util/ConvertToPOJOUtil.java

```diff
@@ -0,0 +1,364 @@
+     */
+    public static<T> List<T> convertDotConnectMapToPOJO(List<Map<String,String>> results, Class classToUse)
+            throws Exception {
+
+        if (Folder.class.equals(classToUse)){
+            return (List<T>) convertDotConnectMapToFolder(results);
+        }
+
+        if (Container.class.equals(classToUse)){
+            return (List<T>) convertDotConnectMapToContainer(results);
+        }
+
+        if (Link.class.equals(classToUse)){
+            return (List<T>) convertDotConnectMapToLink(results);
+        }
+
+        if (Identifier.class.equals(classToUse)){
+            return (List<T>) convertDotConnectMapToIdentifier(results);
+        }
+
+        if (Template.class.equals(classToUse)){
+            return (List<T>) convertDotConnectMapToTemplate(results);
+        }
+
+        List<T> ret;
+        Map<String, String> properties;
+
+        ret = new ArrayList<>();
+
+        if(results == null || results.size()==0){
+            return ret;
+        }
+
+        for (Map<String, String> map : results) {
+            Constructor<?> ctor = classToUse.getConstructor();
+            T object = (T) ctor.newInstance();
+
+            properties = map.keySet().stream().collect(Collectors
+                    .toMap(key -> CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, key), key ->map.get(key)));
+
+            for (String property: properties.keySet()){
+                if (properties.get(property) != null){
+                    if (isFieldPresent(classToUse, String.class, property)){
+                        PropertyUtils.setProperty(object, property, properties.get(property));
+                    }else if (isFieldPresent(classToUse, Integer.TYPE, property)){
+                        PropertyUtils.setProperty(object, property, Integer.parseInt(properties.get(property)));
+                    }else if (isFieldPresent(classToUse, Boolean.TYPE, property)){
+                        PropertyUtils.setProperty(object, property, Boolean.parseBoolean(properties.get(property)));
+                    }else if (isFieldPresent(classToUse, Date.class, property)){
+                        PropertyUtils.setProperty(object, property, df.parse(properties.get(property)));
+                    }else{
+                        Logger.warn(classToUse, "Property " + property + "not set for " + classToUse.getName());
+                    }
+                }
+            }
+
+            ret.add(object);
+        }
+        return ret;
+    }
+
+    /**
+     * Searches a property recursively through classes and superclasses
+     * @param classToUse
+     * @param fieldType
+     * @param property
+     * @return
+     * @throws NoSuchFieldException
+     */
+    private static boolean isFieldPresent(Class classToUse, Class fieldType, String property)
+            throws NoSuchFieldException {
+
+        try{
+            return classToUse.getDeclaredField(property).getType() == fieldType;
+        }catch(NoSuchFieldException e){
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1166)